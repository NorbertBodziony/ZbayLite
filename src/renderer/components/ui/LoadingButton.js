import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import Button from '@material-ui/core/Button'
import CircularProgress from '@material-ui/core/CircularProgress'

const styles = theme => ({
  button: {
    maxWidth: 286,
    height: 60,
    backgroundColor: theme.palette.colors.zbayBlue,
    color: theme.palette.colors.white,
    '&:hover': {
      backgroundColor: theme.palette.colors.zbayBlue
    },
    '&:disabled': {
      backgroundColor: theme.palette.colors.darkGray,
      opacity: 0.7
    }
  },
  progress: {
    color: theme.palette.colors.white
  }
})

export const LoadingButton = ({ classes, inProgress, text, ...other }) => {
  return inProgress ? (
    <Button className={classes.button} {...other}>
      <CircularProgress className={classes.progress} />
    </Button>
  ) : (
    <Button className={classes.button} {...other}>
      {text || 'Continue'}
    </Button>
  )
}

LoadingButton.defaultProps = {
  inProgress: false
}

LoadingButton.propTypes = {
  classes: PropTypes.object.isRequired,
  inProgress: PropTypes.bool,
  finished: PropTypes.bool
}

export default withStyles(styles)(LoadingButton)
