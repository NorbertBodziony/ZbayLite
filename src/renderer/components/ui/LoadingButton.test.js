/* eslint import/first: 0 */
import React from 'react'
import { shallow } from 'enzyme'

import { mockClasses } from '../../../shared/testing/mocks'
import { LoadingButton } from './LoadingButton'

describe('Loading button', () => {
  it('renders component', () => {
    const props = {
      inProgress: false
    }
    const result = shallow(
      <LoadingButton
        classes={mockClasses}
        props={props}
      />
    )
    expect(result).toMatchSnapshot()
  })
})
