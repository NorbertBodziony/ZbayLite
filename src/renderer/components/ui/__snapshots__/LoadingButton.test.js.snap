// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading button renders component 1`] = `
<WithStyles(ForwardRef(Button))
  className="button-mock"
  props={
    Object {
      "inProgress": false,
    }
  }
>
  Continue
</WithStyles(ForwardRef(Button))>
`;
